pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -346,12 +348,13 @@ func (s *ServiceEntryStore) Services() ([]*model.Service, error) {
 		services = append(services, convertServices(cfg)...)
 	}
 
-	return services, nil
+	return autoAllocateIPs(services), nil
 }
 
 // GetService retrieves a service by host name if it exists
 // THIS IS A LINEAR SEARCH WHICH CAUSES ALL SERVICE ENTRIES TO BE RECONVERTED -
 // DO NOT USE
+// NOTE: This does not auto allocate IPs. The service entry implementation is used only for tests.
```

Comment:
I am a bit confused why this is only for tests?

Reply:
I saw some test use the GetService() function. But no other code used it.